### PR TITLE
[MRG] add pytorch model visualization in front-end

### LINKED ISSUE #181: Add pytorch model visualization support
- use [torchviz](https://github.com/szagoruyko/pytorchviz) to render [dot](https://graphviz.org/doc/info/lang.html) from PyTorch model
- Add graph route in fastapi
- create visualization page in frontend, render dot graph into svg format through `graphviz-react` package

### PR DESCRIPTION
**Pull request by univerone**
Closes #181 

#### Before submitting this PR, please make sure you have:

- [x] run `python -m pytest tests/` and confirmed all checks still pass.
- [x] verified that any code or assets from external sources are properly credited.

### CODE REVIEW

File: modelci/types/trtis_objects.py [hunk 1]
```diff
@@ -566,7 +566,7 @@ class ModelOptimizationPolicy(betterproto.Message):
     optimized and prioritized by the backend framework when@@   it is loaded.@@
     """
 
-    # @@  .. cpp:var:: Graph graph@@@@     The graph optimization setting for the
+    # @@  .. cpp:var:: Graph graph.py@@@@     The graph.py optimization setting for the
```
Comment (YuanmingLeee) [inline modelci/types/trtis_objects.py:569]: Please revert changes in this file. Seems like they are irrelevant

**Review by YuanmingLeee — CHANGES_REQUESTED** [review]
Pls update the required environments by adding `torchviz` to `requirement.txt`, `environment.yml` and `pure_reqires.yml`

File: modelci/types/trtis_objects.py (hunk 1 above)
Comment (univerone) [reply modelci/types/trtis_objects.py]: > Please revert changes in this file. Seems like they are irrelevant

OK!

**Comment by YuanmingLeee** [comment]
I realize that `torchviz` claims a dependency on `pytorch`. Since `pytorch` is cuda specified, it should be installed manually after other packages are installed by `conda`.
I will fix this after this merge.

**Review by YuanmingLeee — APPROVED** [review]

**Comment by YuanmingLeee** [comment]
@univerone Change the title to `[MRG}...` if this PR is OK to be merged.

File: frontend/src/pages/Visualization/index.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
+import React from 'react';
```
Comment (HuaizhengZhang) [inline frontend/src/pages/Visualization/index.tsx:1]: the directory  should not be Visualization 

user visualizer

File: frontend/src/routes.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import Dashboard from '@/pages/Dashboard';
 import About from '@/pages/About';
 import Setting from '@/pages/Setting';
 import User from '@/pages/User';
+import Visualization from '@/pages/Visualization';
```
Comment (HuaizhengZhang) [inline frontend/src/routes.ts:6]: rename the function

File: frontend/src/routes.ts [hunk 4]
```diff
@@ -26,6 +27,10 @@ const routerConfig = [
         path: '/setting',
         component: Setting,
       },
+      {
+        path: '/visualization/:id',
+        component: Visualization,
```
Comment (HuaizhengZhang) [inline frontend/src/routes.ts:32]: same as above

File: modelci/app/v1/api.py [hunk 5]
```diff
@@ -8,6 +8,8 @@
 from fastapi import APIRouter
 
 from modelci.app.v1.endpoints import model
+from modelci.app.v1.endpoints import graph
 
 api_router = APIRouter()
 api_router.include_router(model.router, prefix='/model', tags=['model'])
+api_router.include_router(graph.router, prefix='/graph', tags=['graph'])
```
Comment (HuaizhengZhang) [inline modelci/app/v1/api.py:15]: not graph
graph in CS has a very specific meaning

File: modelci/types/vo/model_vo.py [hunk 6]
```diff
@@ -59,6 +59,7 @@ class Engine(CaseInsensitiveEnum):
     TRT = 'TRT'
     TVM = 'TVM'
     CUSTOMIZED = 'Customized'
```
Comment (HuaizhengZhang) [inline modelci/types/vo/model_vo.py:61]: CUSTOMIEZED

File: frontend/src/config.ts [hunk 7]
```diff
@@ -1,5 +1,6 @@
 export default {
     default: {
-        dataURL: 'http://localhost:8000/api/v1/model'
+        modelURL: 'http://localhost:8000/api/v1/model',
+        graphURL: 'http://localhost:8000/api/v1/graph'
```
[the remark this answers is not in the document]
Comment (univerone) [reply frontend/src/config.ts]: OK!

**Review by HuaizhengZhang — DISMISSED** [review]

**Comment by HuaizhengZhang** [comment]
Please update
1. changelog
2. a demo figure (png) to demonstrate the function

Then merge @univerone

**Review by YuanmingLeee — APPROVED** [review]
lgtm

**Comment by univerone** [comment]
> Please update
> 
>     1. changelog
> 
>     2. a demo figure (png) to demonstrate the function
> 
> 
> Then merge @univerone

OK!

**Pull request merged**